raptor_dbw_can/src/DbwNode.cpp

```diff
@@ -848,9 +851,46 @@ void DbwNode::recvCAN(const can_msgs::msg::Frame::SharedPtr msg) {
                 }
             } break;
         }
+
+        // generates TireTemp messages and assign them to TireTempReport.
+        tire_temp_report_msg.front_left = DbwNode::generateTireTemp(fl_tire_temperature);
+        tire_temp_report_msg.front_right = DbwNode::generateTireTemp(fr_tire_temperature);
+        tire_temp_report_msg.rear_left = DbwNode::generateTireTemp(rl_tire_temperature);
+        tire_temp_report_msg.rear_right = DbwNode::generateTireTemp(rr_tire_temperature);
     }
 }
 
+// computes values of TireTemp message for a given array of tire temperatures.
+deep_orange_msgs::msg::TireTemp DbwNode::generateTireTemp(float32[] tire_temperature) {
```

Comment:
Since this function is part of the DBWNode class you should not need to pass in the tire temperature arrays, since they are stored as class members.

Reply:
The reason why I created this function with a temperature array as argument was so that I could swap that array out? Should I create an array containing fl_tire_temperature, fr_tire_temperature, rl_tire_temperature, and rr_tire_temperature in order to loop through them?